bpf/lib/lb.h

```diff
@@ -316,29 +316,9 @@ bool lb6_svc_is_affinity(const struct lb6_service *svc)
 	return svc->flags & SVC_FLAG_AFFINITY;
 }
 
-static __always_inline
-__u8 svc_is_routable_mask(void)
-{
-	__u8 mask = SVC_FLAG_ROUTABLE;
-
-#ifdef ENABLE_LOADBALANCER
-	mask |= SVC_FLAG_LOADBALANCER;
-#endif
-#ifdef ENABLE_NODEPORT
-	mask |= SVC_FLAG_NODEPORT;
-#endif
-#ifdef ENABLE_EXTERNAL_IP
-	mask |= SVC_FLAG_EXTERNAL_IP;
-#endif
-#ifdef ENABLE_HOSTPORT
-	mask |= SVC_FLAG_HOSTPORT;
-#endif
-	return mask;
-}
-
 static __always_inline bool __lb_svc_is_routable(__u8 flags)
 {
-	return (flags & svc_is_routable_mask()) > SVC_FLAG_ROUTABLE;
```

Comment:
Shouldn't this be guarded by `EnableClusterIPExternalAccess` config?

Reply:
Not sure what you mean by guarded. The flag is disabled by default in which case `SVC_FLAG_ROUTABLE` is unset for ClusterIP services. Earlier `SVC_FLAG_ROUTABLE` was set for ClusterIP services, but higher bits were unset and hence this function returned false for `ClusterIP` services.

Reply:
Speaking of which, I can't find where, in your PR, `SVC_FLAG_ROUTABLE` would be set or unset depending on the value for `config.ExternalClusterIP`. Shouldn't there be a change to the definition of `SVC_FLAG_ROUTABLE` in bpf/lib/common.h?

Reply:
The change to this logic is in the agent in `updateMasterService`: https://github.com/cilium/cilium/pull/15650/files#diff-8eff0d99dd1ceb7d15ce632811672e7b17a17fb5e984fafa7875d6ad2433b3d8R519. It was already set earlier and in this PR I'm turning it off for ClusterIP services unless the new external access flag is set.

Reply:
Ok thank you. On my first read, I somehow understood that you would be changing the value for `SVC_FLAG_ROUTABLE` depending on the value of `ExternalClusterIP`, but I understand this is not the case - You're just changing the `flags`.